dlrover/trainer/tensorflow/executor/base_executor.py

```diff
@@ -0,0 +1,168 @@
+# Copyright 2023 The DLRover Authors. All rights reserved.
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import json
+import os
+
+import tensorflow.compat.v1 as tf
+from tensorflow.core.protobuf import cluster_pb2
+from tensorflow.python.training import server_lib
+
+from dlrover.trainer.util.log_util import default_logger as logger
+
+tf.disable_v2_behavior()
+
+
+class BaseExecutor:
+    def __init__(self):
+        self.cluster_spec = None
+        self.task_id = None
+        self.task_type = None
+        self.address = None
+        self.role = None
+        self.cluster_spec = None
+        self.mini_cluster_spec = None
```

Comment:
What's the difference between `self.cluster_spec` and `self.mini_cluster_spec`.

Reply:
`self.cluster_spec` is the full descripation of the training clustser which includes all parameter servers, evaluator and workers. 
`self.mini_cluster_spec` is the descripation of the training clustser which includes all parameter servers and the worker itself.

self.mini_cluster_spec is used for dynamic networking since in this case it doesn't need to interact with other workers.

For example, let's assume that the training cluster is` {"cluster": {"chief": ["job0.chief0.fs.rpc:0"], "evaluator": ["job0.evaluator0.fs.rpc:0"],  "ps": ["11.250.95.69:55145"], "worker": ["job0.worker0.fs.rpc:0", "job0.worker1.fs.rpc:0"]}}`

For worker-0，self.cluster_spec is `{"cluster": {  "chief": ["job0.chief0.fs.rpc:0"], "evaluator": ["job0.evaluator0.fs.rpc:0"],  "ps": ["11.250.95.69:55145"], "worker": ["job0.worker0.fs.rpc:0"]}}, and self.minicluster_spec is {"cluster": {"ps": ["11.250.95.69:55145"], "worker": ["job0.worker0.fs.rpc:0"]}}`